src/domain/chainContext.ts

```diff
@@ -209,11 +220,30 @@ export class ChainContext {
         _(`Block ${blockNumber} has been processed.`);
       } catch (error) {
         console.error(
-          `[runBlockWatcher:chainId:${chainId}] Error in processBlock`,
+          `[runBlockWatcher:chainId:${chainId}] Error in pollContractForEvents for block ${blockNumber}`,
           error
         );
       }
     });
+
+    // We run a watchdog to check if we are receiving blocks.
+    while (true) {
+      // sleep for 5 seconds
+      await new Promise((resolve) => setTimeout(resolve, WATCHDOG_FREQUENCY));
+      const currentTime = new Date().getTime();
+      const timeElapsed = currentTime - timeLastBlockProcessed;
+      // Todo: set this as trace to avoid spamming the logs
+      console.log(
+        `[runBlockWatcher:chainId:${chainId}] diff: ${timeElapsed}ms`
+      );
+
+      // If we haven't received a block in 30 seconds, restart the watcher
+      if (timeElapsed >= WATCHDOG_KILL_THRESHOLD) {
+        console.error(`[runBlockWatcher:chainId:${chainId}] Watchdog timeout`);
+        await registry.storage.close();
+        process.exit(1);
+      }
```

Comment:
Comment doesn't match the code.
Auto-restarting might be a side effect, but the logic to restart it is not here.

Reply:
Done.